### ISSUE TITLE
Custom matrix conflicts with -g

### ISSUE THREAD

**Issue by jcoupey** [VROOM-Project/vroom#87]
The `-g` command-line flag results in a `route` request to OSRM after optimization to retrieve additional information such as distance, route geometry etc.

When solving a problem while providing a custom matrix, this does not necessarily make sense any more as jobs can be provided without a `location`. The problem here is that in that case we hit an assertion about missing coordinates:

```bash
vroom: structures/vroom/location.cpp:43: coordinate_t location_t::lat() const: Assertion `this->has_coordinates()' failed.
```

We should at least handle the error gracefully and return an explicit message in that case.

**Comment by jcoupey** [comment]
I see two ways of dealing with this:

1. disable `-g` altogether with custom matrices
2. error on `-g` use with custom matrices when not all jobs have a `location`

Option 2. would allow to still use a post-optimization `route` request to OSRM, with no guarantee of consistency between custom matrix and OSRM costs. Not sure if there is a strong use-case for that?

**Comment by jcoupey** [comment]
Going for option 2 as it's still a very simple fix and the consistency concern is not that big:
- `cost` and duration-related keys (`arrival`, `duration` etc.) are clearly separated in the API
- the additional `route` request has to be explicitly fired using `-g`
- people using custom matrices should know what they are doing in the first place.

**Comment by jcoupey** [comment]
Fixed in #105.
